PhysicsTools/PatAlgos/plugins/PATMuonProducer.cc

```diff
@@ -1094,30 +1096,55 @@ void PATMuonProducer::embedHighLevel(pat::Muon& aMuon,
                                      bool primaryVertexIsValid,
                                      reco::BeamSpot& beamspot,
                                      bool beamspotIsValid) {
+  // Generic variable to store measurements
+  std::pair<bool,Measurement1D> result;
+
   // Correct to PV
 
   // PV2D
-  std::pair<bool, Measurement1D> result =
-      IPTools::signedTransverseImpactParameter(tt, GlobalVector(track->px(), track->py(), track->pz()), primaryVertex);
-  double d0_corr = result.second.value();
-  double d0_err = primaryVertexIsValid ? result.second.error() : -1.0;
-  aMuon.setDB(d0_corr, d0_err, pat::Muon::PV2D);
+  if (getdBFromTrack_){
+    aMuon.setDB(track->dxy(primaryVertex.position()),
+                track->dxyError(primaryVertex.position(), primaryVertex.covariance()),
+                pat::Muon::PV2D);
+  }
+  else{
+    result =
+      IPTools::signedTransverseImpactParameter(tt,
+                                               GlobalVector(track->px(),
+                                                            track->py(),
+                                                            track->pz()),
+                                               primaryVertex);
```

Comment:
were these linebreaks triggered by some format run? (it looks like master would have it in one line)
```suggestion
      IPTools::signedTransverseImpactParameter(tt, GlobalVector(track->px(), track->py(), track->pz()), primaryVertex);
```

Reply:
I think it was my fault due to copying from the github raw format to my text editor. They don't seem to mix well.